codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -599,14 +601,28 @@ void resizeIfNecessary(int index) {
                 }
             }
         }
+
+        @Override
+        public boolean equals(final Object obj) {
+            return super.equals(obj);
+        }
+
+        @Override
+        public int hashCode() {
+            long value = identity;
+            if (value == 0) {
+                return System.identityHashCode(this);
+            }
+            return (int) (value ^ (value >>> 32));
```

Comment:
Consider caching this value in ctor instead of `identity` to avoid re-calculation on every invocation.

Reply:
IDK if we can benefit that much about it (ie I should benchmark it), given that we tends to create tons of stream and is a pretty cheap computation I would leave it as it is...
The major bottleneck is still reading from a field, and it is going to happen the same, in both cases